edweiss-app/components/core/calendar.tsx

```diff
@@ -57,102 +60,190 @@ export const Calendar = ({ courses, type }: { courses: { id: string; data: Cours
-                </TTouchableOpacity>
+            {/* Display current date */}
+            {format === 'day' && (
+                <TView pt={35} backgroundColor="constantBlack" alignItems="center">
+                    <TText color="text" m={5} align="center" size="lg" style={{ width: '86%' }}>
+                        {stringDate}
+                    </TText>
+                </TView>
             )}
 
-            {format === "week" && (
+            {/* Week view */}
+            {format === 'week' && (
                 <TView flexDirection="row" >
-                    <TText align='center' style={{ width: '7%' }} >Time</TText>
-                    {DAY_NAMES.map((dayName) => (
-                        <TText key={dayName} align='center' style={{ width: '13.3%' }}>
-                            {dayName}
+                    <TView style={{ width: '5%' }} />
+                    {weekDates.map(weekDate => (
+                        <TText
+                            pt={20}
+                            key={weekDate.toISOString()}
+                            align="center"
+                            style={{ width: '13.3%' }}
+                        >
+                            {weekDate.toLocaleDateString('fr-FR', {
+                                weekday: 'short',
+                                day: 'numeric',
+                                month: 'short',
+                            })}
                         </TText>
                     ))}
                 </TView>
             )}
+
             <ScrollView
                 ref={scrollViewRef}
                 style={{ flex: 1 }}
-                showsVerticalScrollIndicator={true}>
+                showsVerticalScrollIndicator={true}
+            >
                 <TView flexDirection="row">
-                    <TView style={(format == 'week' && { width: '7%' }) || (format == 'day' && { width: '14%' })}>
+                    {/* Render time labels */}
+                    <TView style={format === 'week' ? { width: '5%' } : { width: '14%' }}>
                         {Array.from({ length: TOTAL_HOURS }).map((_, hour) => (
-                            <TView key={`hour-${new Date().getTime()}-${hour}`} alignItems='center' justifyContent='center' bt={1} bb={1} borderColor='yellow' style={{ height: HOUR_BLOCK_HEIGHT }}>
-                                <TText color='text' size={12}>{`${hour}:00`}</TText>
+                            <TView
+                                key={`hour-${date.getTime()}-${hour}`}
+                                alignItems="center"
+                                justifyContent="center"
+                                bt={1}
+                                bb={1}
+                                borderColor="overlay0"
+                                backgroundColor="constantBlack"
+                                style={{ height: HOUR_BLOCK_HEIGHT }}
+                            >
```

Comment:
Firstly I would like to say that you did a great job in this file, it was not easy but great job. As when I was testing your code I realize that it was a bit slow. I would like to share with you some ways to make it faster.

The `renderDay` function currently creates a new array of hour blocks every time it is invoked. While it works, this approach can lead to unnecessary re-renders, especially when the calendar handles multiple days or if the data being passed changes frequently.

Here want I am proposing :
```tsx
const hoursArray = useMemo(() => Array.from({ length: TOTAL_HOURS }), []);

{hoursArray.map((_, hour) => {
    // Hourly rendering logic...
})}
```
This way you could:
-  Reduces memory allocation overhead.
- Improves rendering efficiency, especially for large data sets or when navigating between days/weeks.

I hope that a bit more clear, tell me if you need more help getting this set up.